### ISSUE TITLE
Deprecating node-redis in favour of ioredis

### ISSUE THREAD

**Issue by chasemcdo** [hwchase17/langchainjs#2037]
Hey @jacoblee93 
Wanted to compile a list of next steps following:
- https://github.com/hwchase17/langchainjs/pull/1920
- https://github.com/hwchase17/langchainjs/issues/1710

And I can help pick away at the migration. Most should be fairly straightforward.

Item to deprecate:
- [Original Redis Message History](https://github.com/hwchase17/langchainjs/blob/2e63ee0a4e685b43f420d79d7c38ba6d16e374fa/langchain/src/stores/message/redis.ts)

Items to migrate:
- [Redis Cache](https://github.com/hwchase17/langchainjs/blob/2e63ee0a4e685b43f420d79d7c38ba6d16e374fa/langchain/src/cache/redis.ts)
- [Redis VectorStore](https://github.com/hwchase17/langchainjs/blob/2e63ee0a4e685b43f420d79d7c38ba6d16e374fa/langchain/src/vectorstores/redis.ts)
- All Examples (most of which should still be compatible as they use the standard URL passing)

Are there any other items I'm missing?

**Comment by jacoblee93** [comment]
Thank you! I think it's just those two.

**Comment by chasemcdo** [comment]
For these deprecations do we want to follow a similar setup to what we did with the messages?
- New ioredis import
- Deprecation label on old node-redis version

- in 0.1.0 release we'll move the contents of the ioredis import into redis and add the deprecation to the ioredis path?

**Comment by jacoblee93** [comment]
I think we just deprecate *all* the Redis classes at the same time once everything's been replaced - ideally people will only use one package for all Redis-related tasks since the functionality should be a superset.

Then in 0.1.0 we cut all the old stuff.

**Comment by chasemcdo** [comment]
Yes still incomplete. The VectoreStore is the last remaining item as well as some potential examples I may have missed

**Comment by chasemcdo** [comment]
Hey @dosubot! Yes this is still applicable to the latest version of langchainjs; however, it looks like the `ioredis` library may be missing some vector store components.

**Comment by chasemcdo** [comment]
I think it is good to mark this as complete, since it doesn't look like ioredis presently supports vector stores, so this would be blocked by that support being added with no clear roadmap on a timeline.
